example/basic/basic_hcube.cxx

```diff
@@ -26,81 +26,137 @@
 #include <t8_forest.h>
 #include <t8_cmesh_vtk.h>
 
+/** Construct the cmesh given the dimension of the examples
```

Comment:
What kind of cmesh is constructed?
How do the parameters affect how the cmesh looks like?